src/Get-BicepDeployments.ps1

```diff
@@ -129,38 +130,28 @@ $deploymentObjects = foreach ($deployment in $deployments) {
                 Write-Debug "[$deploymentName][$environmentName] No files used by the deployment have been modified. Deployment not included."
             }
         }
+        else {
+            Write-Debug "[$deploymentName][$environmentName] Skipping modified files check. GitHub event is `"$($EventName)`". All deployments included by default."
+        }
 
         #* Pattern filter
-        Write-Debug "[$deploymentName][$environmentName] Checking if deployment matches pattern filter."
-        if ($Pattern) {
-            if ($deploymentObject.Name -match $Pattern) {
-                $deploymentObject.Deploy = $true
-                Write-Debug "[$deploymentName][$environmentName] Pattern [$Pattern] matched successfully. Deployment included."
+        if ($deploymentObject.Deploy) {
```

Comment:
$deploymentObject.Deploy is checked several times throughout this file at the same scope. Is this something we could simplify/refactor? (perhaps in a separate PR since this is a bugfix)

Reply:
This is checked multiple times to improve the reading experience in the logs. 
Each of the code blocks have the possibility to disable the deployment. All blocks/filters needs to pass for the deployment to be valid. 

By checking this before each of the blocks, we can skip processing the block and rather print a well formed message stating that the filter check is skipped since the deployment is already disabled.